npk/src/npk.rs

```diff
@@ -459,66 +461,72 @@ pub fn pack_with(
         .with_extension(&NPK_EXT);
-    let npk = File::create(&npk_dest).map_err(|e| Error::Os {
-        context: format!("Failed to create NPK at '{}'", &npk_dest.display()),
+    let npk = File::create(&npk_dest).await.map_err(|e| Error::Os {
+        context: format!("Failed to create NPK: '{}'", &npk_dest.display()),
         error: e,
     })?;
-    builder.build(&npk)
+    builder.build(npk.into_std().await).await
 }
 
-pub fn unpack(npk: &Path, out: &Path) -> Result<(), Error> {
-    let mut zip = open_zipped_npk(&npk)?;
+pub async fn unpack(npk: &Path, out: &Path) -> Result<(), Error> {
+    let mut zip = open_zip(&npk).await?;
     zip.extract(&out).map_err(|e| Error::Zip {
         context: format!("Failed to extract NPK to '{}'", &out.display()),
         error: e,
     })?;
     let fsimg = out.join(&FS_IMG_NAME);
-    unpack_squashfs(&fsimg, &out)
+    unpack_squashfs(&fsimg, &out).await
 }
 
 /// Generate a keypair suitable for signing and verifying NPKs
-pub fn gen_key(name: &str, out: &Path) -> Result<(), Error> {
+pub async fn gen_key(name: &str, out: &Path) -> Result<(), Error> {
     let mut csprng = OsRng {};
     let key_pair = Keypair::generate(&mut csprng);
     let pub_key = out.join(name).with_extension("pub");
     let prv_key = out.join(name).with_extension("key");
-    assume_non_existing(&pub_key)?;
-    assume_non_existing(&prv_key)?;
+    assume_non_existing(&pub_key).await?;
+    assume_non_existing(&prv_key).await?;
 
-    fn write(data: &[u8], path: &Path) -> Result<(), Error> {
-        let mut file = File::create(&path).map_err(|e| Error::Os {
+    async fn write(data: &[u8], path: &Path) -> Result<(), Error> {
+        let mut file = File::create(&path).await.map_err(|e| Error::Os {
             context: format!("Failed to create '{}'", &path.display()),
             error: e,
         })?;
 
-        file.write_all(&data).map_err(|e| Error::Os {
+        file.write_all(&data).await.map_err(|e| Error::Os {
             context: format!("Failed to write to '{}'", &path.display()),
             error: e,
         })?;
         Ok(())
     }
-    write(&key_pair.public.to_bytes(), &pub_key)?;
-    write(&key_pair.secret.to_bytes(), &prv_key)?;
+    write(&key_pair.public.to_bytes(), &pub_key).await?;
+    write(&key_pair.secret.to_bytes(), &prv_key).await?;
     Ok(())
 }
 
-pub fn open_zipped_npk(npk: &Path) -> Result<ZipArchive<File>, Error> {
-    let zip = zip::ZipArchive::new(File::open(&npk).map_err(|e| Error::Os {
-        context: format!("Failed to open NPK at '{}'", &npk.display()),
-        error: e,
-    })?)
+pub async fn open_zip(file: &Path) -> Result<ZipArchive<std::fs::File>, Error> {
+    let zip = zip::ZipArchive::new(
+        File::open(&file)
+            .await
+            .map_err(|e| Error::Os {
+                context: format!("Failed to open file: '{}'", &file.display()),
+                error: e,
+            })?
+            .into_std()
+            .await,
+    )
```

Comment:
I think `zip::ZipArchive::new` is still a synchronous function and it should be wrapped in a `task::block_in_place` or a `task::spawn_blocking`. Maybe consider:

```rust
pub async fn open_zip(file: &Path) -> Result<ZipArchive<std::fs::File>, Error> {
    let path = file.to_owned();
    task::block_in_place(move || {
        let file = std::fs::File::open(&path).map_err(|e| Error::Os {
            context: format!("Failed to open file: '{}'", &file.display()),
            error: e,
        })?;
        zip::ZipArchive::new(file).map_err(|e| Error::Zip {
            context: format!("Failed to parse ZIP format: '{}'", path.display()),
            error: e,
        })
    })
}
```